database/sql/headers.go

```diff
@@ -473,12 +472,9 @@ func (h *HeadersDb) getMerkleRootConfirmation(item domains.MerkleRootConfirmatio
 	var hash sql.NullString
 	err := h.db.Get(&hash, sqlVerifyHash, item.MerkleRoot, item.BlockHeight)
 
-	if err != nil {
-		if errors.Is(err, sql.ErrNoRows) {
-		} else {
-			return nil, err
-		}
-	} else {
+	if err != nil && !errors.Is(err, sql.ErrNoRows) {
+		return nil, err
+	} else if err == nil {
```

Comment:
same bad readability and finishing `if` block with `else if` not good idea
what about this 
```go
if err != nil {
    if !errors.Is(err, sql.ErrNoRows) {
        return nil, err
    }
}

// At this point, err is either nil or sql.ErrNoRows, handle the nil case 
if err == nil {
    confirmation.Hash = hash
}
return confirmation, nil  
```

Reply:
For me, your proposition has still readability issues. I will look for something better. But for me it's matter of taste, and probably we can discuss this for hours and won't came into a conclusion.